bundles/org.openhab.binding.matter/src/test/java/org/openhab/binding/matter/internal/controller/devices/converter/ColorControlConverterTest.java

```diff
@@ -0,0 +1,226 @@
+        converter.supportsHue = true;
+
+        AttributeChangedMessage levelMsg = new AttributeChangedMessage();
+        levelMsg.path = new Path();
+        levelMsg.path.attributeName = "currentLevel";
+        levelMsg.value = 254; // 100%
+        converter.onEvent(levelMsg);
+
+        AttributeChangedMessage modeMsg = new AttributeChangedMessage();
+        modeMsg.path = new Path();
+        modeMsg.path.attributeName = "colorMode";
+        modeMsg.value = ColorModeEnum.CURRENT_HUE_AND_CURRENT_SATURATION;
+        converter.onEvent(modeMsg);
+
+        AttributeChangedMessage hueMsg = new AttributeChangedMessage();
+        hueMsg.path = new Path();
+        hueMsg.path.attributeName = "currentHue";
+        hueMsg.value = 127; // ~180 degrees
+        converter.onEvent(hueMsg);
+
+        AttributeChangedMessage satMsg = new AttributeChangedMessage();
+        satMsg.path = new Path();
+        satMsg.path.attributeName = "currentSaturation";
+        satMsg.value = 254; // 100%
+        converter.onEvent(satMsg);
+
+        // Wait for color update timer
+        // Thread.sleep(600); // Wait slightly longer than the 500ms timer
+
+        verify(mockHandler, times(1)).updateState(eq(1), eq("colorcontrol-color"),
+                eq(new HSBType(new DecimalType(180), new PercentType(100), new PercentType(100))));
+    }
+
+    @Test
+    void testOnEventWithColorTemperature() throws InterruptedException {
+        converter.supportsColorTemperature = true;
+
+        AttributeChangedMessage modeMsg = new AttributeChangedMessage();
+        modeMsg.path = new Path();
+        modeMsg.path.attributeName = "colorMode";
+        modeMsg.value = ColorModeEnum.COLOR_TEMPERATURE_MIREDS;
+        converter.onEvent(modeMsg);
+
+        AttributeChangedMessage tempMsg = new AttributeChangedMessage();
+        tempMsg.path = new Path();
+        tempMsg.path.attributeName = "colorTemperatureMireds";
+        tempMsg.value = 250;
+        converter.onEvent(tempMsg);
+
+        // Wait for color update timer
+        Thread.sleep(600);
+
+        verify(mockHandler, times(1)).updateState(eq(1), eq("colorcontrol-temperature"), eq(new PercentType(27)));
+        verify(mockHandler, times(1)).updateState(eq(1), eq("colorcontrol-temperature-abs"),
+                eq(new QuantityType<>(250, Units.MIRED)));
+    }
+
+    @Test
+    void testInitState() throws InterruptedException {
+        mockColorCluster.colorMode = ColorModeEnum.CURRENT_HUE_AND_CURRENT_SATURATION;
+        mockColorCluster.currentHue = 127; // ~180 degrees
+        mockColorCluster.currentSaturation = 254; // 100%
+        mockColorCluster.colorTemperatureMireds = 250;
+        mockColorCluster.featureMap.hueSaturation = true;
+        mockColorCluster.featureMap.colorTemperature = true;
+        mockLevelCluster.currentLevel = 254;
+        converter.initState(true, mockLevelCluster);
+
+        // Wait for color update timer
+        Thread.sleep(600);
```

Comment:
And here? I put a comment some time ago with a concern about these, but I don't remember if I commented them all, and what the outcome/response was.